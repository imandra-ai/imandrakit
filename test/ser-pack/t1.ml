type foo = {
  a: int;
  b: float;
}
[@@deriving serpack, show]

let s = Imandrakit_ser_pack.to_cbor_string foo_to_serpack { a = 1; b = 2.0 };;

Printf.printf "serialized to %S\n" s;;
Printf.printf "len: %d\n" (String.length s)

let foo2 = Imandrakit_ser_pack.of_cbor_string_exn foo_of_serpack s;;

Format.printf "foo after roundtrip: %a@." pp_foo foo2
