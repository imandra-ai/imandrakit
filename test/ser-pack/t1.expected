serialized to "\162ak\198\000ah\129\162\000\001\001\251@\000\000\000\000\000\000\000"
len: 21
foo after roundtrip: { T1.a = 1; b = 2. }
